toc/rfc/rfc-draft-repository-ownership.md

```diff
@@ -0,0 +1,78 @@
+# Meta
+[meta]: #meta
+- Name: Working Group Repository Ownership
+- Start Date: 2022-04-19
+- Author(s): [@emalm](https://github.com/emalm)
+- Status: Draft <!-- Acceptable values: Draft, Approved, On Hold, Superseded -->
+- RFC Pull Request: [community#251](https://github.com/cloudfoundry/community/pull/251)
+
+
+## Summary
+
+Adding a repository to the ownership scope of a Working Group, removing a repository from its scope, or transferring a repository between Working Groups within the `cloudfoundry` GitHub organization requires approval separately from each affected Working Group and from the Technical Oversight Committee (TOC).
+
+
+## Problem
+
+As the CF community completes its rollout of the new TOC and Working Group governance structure, the large number of unaddressed PRs about repository ownership across Working Groups has illustrated that the community needs a clearer and more consistent process to handle such ownership changes. In particular, there are currently a significant number of repositories in the [`cloudfoundry`](https://github.com/cloudfoundry) and [`cloudfoundry-incubator`](https://github.com/cloudfoundry-incubator) GitHub organizations that are not yet officially assigned to a Working Group, and the community must decide which ones will be managed by which Working Groups and which ones will be archived.
+
+Even after the CF community resolves the ownership of the existing repositories in these two GitHub organizations, it still needs a clear process to create new repositories under a Working Group, to move repositories between Working Groups, and to remove a repository from a Working Group.
+
+
+## Proposal
+
+### Ownership model
+
+We can represent the ownership structure of the repositories in the `cloudfoundry` GitHub organization as a partition of the set of repositories. Each Working Group has a subset of repositories that it owns, and there are separate subsets under ownership of the TOC and of the CF Foundation staff, respectively. Finally, there are two separate subsets of unmanaged repositories: ones that are archived and ones that are active. Since this collection of subsets is a partition, the subsets MUST be mutually exclusive and comprehensively exhaustive: each repository in the organization is in exactly one of these subsets.
+
+The subset of repositories that each Working Group owns MUST be identical to the set of repositories listed in its charter description. This set of repositories MAY further be partitioned into technical areas within the Working Group, at the discretion of the Working Group.
+
+The subsets of repositories that the TOC and the CF Foundation staff own SHOULD be as small as possible and should be scoped to those that enable the effective operation of the CF community. They SHOULD NOT be concerned with development of tools and systems that are broadly valuable to the end-user community, as the Working Groups collectively SHOULD own these kinds of repositories.
```

Comment:
I think we should list the TOC owned repos e.g. in https://github.com/cloudfoundry/community/blob/main/toc/TOC.md in the same format as in WG charters (or in a separate WG charter like document). This makes the list explicit and allows to apply the same/similar processes and automation.
So far we talk only about this community repo, isn't it?

Reply:
Sure, that makes sense. I think the TOC.md file counts as the charter document for the TOC (apart from the real definition of the TOC scope in the Foundation charter document), so that seems like a good place to list both sets of repositories. Pushed https://github.com/cloudfoundry/community/pull/251/commits/8e86f682d29040125a12ce4c272f1a6a5a0f1412 to adjust this language.